cli/src/commands/split.rs

```diff
@@ -208,3 +201,37 @@ the operation will be aborted.
     tx.finish(ui, format!("split commit {}", commit.id().hex()))?;
     Ok(())
 }
+
+// Rebases the children of `original_commit` by replacing `original_commit` with
+// `new_siblings`. Any parents other than `original_commit` will remain after
+// the rebase.
+fn rebase_children_for_siblings_split(
+    tx: &mut WorkspaceCommandTransaction,
+    settings: &UserSettings,
+    original_commit: &Commit,
+    new_siblings: Vec<Commit>,
+) -> Result<usize, CommandError> {
+    let children: Vec<Commit> = RevsetExpression::commit(original_commit.id().clone())
+        .children()
+        .evaluate_programmatic(tx.base_repo().as_ref())?
+        .iter()
+        .commits(tx.base_repo().store())
+        .try_collect()?;
+    let mut num_rebased = 0;
+    for child in children {
+        let new_parents = child
+            .parents()
+            .into_iter()
+            .flat_map(|c| {
+                if c.id() == original_commit.id() {
+                    new_siblings.clone().into_iter()
+                } else {
+                    vec![c].into_iter()
```

Comment:
nit: I think this is `std::iter::once(c)`

Reply:
The types of the two iterators need to be the same, otherwise it won't compile. I suppose we could box two different types of iterators, but that also requires an explicit `-> Box<dyn Iterator<Item=Commit>>` on the lambda.

It turns out that the `into_iter()` calls aren't necessary though so I removed them.